src/rgw/rgw_sal_store.h

```diff
@@ -27,6 +27,13 @@ class StoreDriver : public Driver {
     virtual uint64_t get_new_req_id() override {
       return ceph::util::generate_random_number<uint64_t>();
     }
+
+    int read_topics(const std::string& tenant, rgw_pubsub_topics& topics, RGWObjVersionTracker* objv_tracker,
+        optional_yield y, const DoutPrefixProvider *dpp) override {return 0;}
+    int write_topics(const std::string& tenant, const rgw_pubsub_topics& topics, RGWObjVersionTracker* objv_tracker,
+	optional_yield y, const DoutPrefixProvider *dpp) override {return 0;}
+    int remove_topics(const std::string& tenant, RGWObjVersionTracker* objv_tracker,
+        optional_yield y, const DoutPrefixProvider *dpp) override {return 0;}
```

Comment:
i think it's better to return an error like `-ENOTSUP` than success, especially for `write_topics()` where success implies we can read them back

Reply:
ok. will return `EOPNOTSUPP` for the write.
for the read i will return `ENOENT` (just to make sure we don't break any code path that tries to read notifications)
for the remove, can be either success or `EOPNOTSUPP` - not sure which is better.

Reply:
> ok. will return `EOPNOTSUPP` for the write.
> for the read i will return `ENOENT` (just to make sure we don't break any code path that tries to read notifications)

:+1: 

> for the remove, can be either success or `EOPNOTSUPP` - not sure which is better.

i would suggest `ENOENT` over success